app/internal/agenda_events.py

```diff
@@ -0,0 +1,54 @@
+from datetime import date, datetime, timedelta
+from typing import Optional
+
+from app.database.models import Event
+from app.database.database import SessionLocal
+
+from sqlalchemy.exc import SQLAlchemyError
+
+
+def get_events_per_dates(session: SessionLocal, user_id: int, start: Optional[date], end: Optional[date]) -> list:
```

Comment:
Consider changing `list` to `List[DataType]`, where DataType is the type of the variables in the list.